exonum/src/api/node/public/explorer.rs

```diff
@@ -276,6 +293,59 @@ impl ExplorerApi {
             })
     }
 
+    /// Return call status of committed transaction.
+    pub fn transaction_status(
+        snapshot: &dyn Snapshot,
+        query: TransactionQuery,
+    ) -> Result<CallStatusResponse, ApiError> {
+        let explorer = BlockchainExplorer::new(snapshot);
+
+        let tx_info = explorer.transaction(&query.hash).ok_or_else(|| {
+            ApiError::NotFound(format!("Unknown transaction hash ({})", query.hash))
+        })?;
+
+        let tx_info = match tx_info {
+            TransactionInfo::Committed(info) => info,
+            TransactionInfo::InPool { .. } => {
+                let err = ApiError::NotFound(format!(
+                    "Requested transaction ({}) is not executed yet",
+                    query.hash
+                ));
+                return Err(err);
+            }
+        };
+
+        let call_in_block = CallInBlock::transaction(tx_info.location().position_in_block());
+        let block_height = tx_info.location().block_height();
+
+        let status = ExecutionStatus(explorer.call_status(block_height, call_in_block));
+        Ok(CallStatusResponse { status })
+    }
+
+    /// Return call status of `before_transactions` hook.
+    pub fn before_transactions_status(
```

Comment:
Nit: you can provide a type of the hook as an argument from `wire` method, and reduce `before_transactions_status` / `after_transactions_status` to one function, e.g. `hook_status`.